InsightLogParser.Client/Spider.cs

```diff
@@ -150,6 +150,14 @@ public async Task SolvedAsync(DateTimeOffset timestamp, int puzzleId, PuzzleZone
             }
 
             _messageWriter.WriteEndSolved();
+
```

Comment:
This section "teleports" the player to the puzzle they just solved.  NextRouteWaypoint only handled cases correctly when you solved the previous puzzle in the route... using next on the route used the previous puzzle in the route, which is incorrect if you skipped some puzzles and teleported since your last solve.

This also has the nice added benefit of updating your targeting if you solve puzzles in transit.

You may want to consider skipping this code for puzzles that aren't solved at their location, ie: Wandering Echoes & Matchboxes, but I left that in for now.